lib/fluent/plugin/in_s3.rb

```diff
@@ -0,0 +1,285 @@
+module Fluent
+  require 'fluent/mixin/config_placeholders'
+
+  class S3Input < Input
+    Fluent::Plugin.register_input('s3', self)
+
+    def initialize
+      super
+      require 'aws-sdk-resources'
+      require 'zlib'
+      require 'time'
+      require 'tempfile'
+
+      @extractor = nil
+    end
+
+    config_param :use_server_side_encryption, :string, :default => nil
+    config_param :aws_key_id, :string, :default => nil, :secret => true
+    config_param :aws_sec_key, :string, :default => nil, :secret => true
+    config_section :assume_role_credentials, :multi => false do
+      config_param :role_arn, :string
+      config_param :role_session_name, :string
+      config_param :policy, :string, :default => nil
+      config_param :duration_seconds, :integer, :default => nil
+      config_param :external_id, :string, :default => nil
+    end
+    config_section :instance_profile_credentials, :multi => false do
+      config_param :retries, :integer, :default => nil
+      config_param :ip_address, :string, :default => nil
+      config_param :port, :integer, :default => nil
+      config_param :http_open_timeout, :float, :default => nil
+      config_param :http_read_timeout, :float, :default => nil
+      # config_param :delay, :integer or :proc, :default => nil
+      # config_param :http_degub_output, :io, :default => nil
+    end
+    config_section :shared_credentials, :multi => false do
+      config_param :path, :string, :default => nil
+      config_param :profile_name, :string, :default => nil
+    end
+    config_param :s3_bucket, :string
+    config_param :s3_region, :string, :default => ENV["AWS_REGION"] || "us-east-1"
+    config_param :store_as, :string, :default => "gzip"
+    config_param :check_apikey_on_start, :bool, :default => true
+    config_param :proxy_uri, :string, :default => nil
+    config_param :format, :string, :default => 'none'
+
+    config_section :sqs, :required => true, :multi => false do
+      config_param :queue_name, :string, :default => nil
+      config_param :skip_delete, :bool, :default => false
+      config_param :wait_time_seconds, :integer, :default => 20
+    end
+
+    config_param :tag, :string, :default => "input.s3"
+
+    attr_reader :bucket
+
+    def configure(conf)
+      super
+
+      unless @sqs.queue_name
+        raise ConfigError, "sqs/queue_name is required"
+      end
+
+      begin
+        @extractor = EXTRACTOR_REGISTRY.lookup(@store_as).new(log: log)
+      rescue
+        $log.warn "#{@store_as} not found. Use 'text' instead"
```

Comment:
In s3 output, we use this fallback for backward compatibility.
I think raising error is better in s3 input.